WordPress/gutenberg
# [RNMobile] Remove third-party dependency on react-native-hsv-color-picker

Related:

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/5847
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6029

Todo:
- [x] Add integration tests to Color Picker
- [x] Migrate tinycolor2 references to use colord library


## What?
Replaces the forked version of [react-native-hsv-color-picker](https://github.com/wordpress-mobile/react-native-hsv-color-picker/) in favor of implementing the library within Gutenberg instead.

## Why?
To reduce the number of third-party libraries and dependencies in order to better support React Native's [New Architecture](https://reactnative.dev/docs/new-architecture-intro) and simplify the React Native upgrade process for Gutenberg Mobile.

## How?
The PR takes the source of our forked react-native-hsv-color-picker code, and reimplements the source within the Color Picker component. The new HSV components (`HuePicker`, `SaturationValuePicker`) are just UI interfaces that leverage the libraries `tinycolor2` and `react-native-linear-gradient`, which are conveniently already dependencies of the project.

## Testing Instructions
1. Create a post
2. Add a block that implements the color picker (like Cover block, and select the gradient option)
3. Observe that the Color Picker behavior has not changed

## Screenshots or screencast 

<video src="https://github.com/WordPress/gutenberg/assets/643285/1601ef5c-13ec-41f9-92c2-8e01a83b21ac" />
